android-library/src/main/java/com/zetapush/library/KeyValueCredentialsStorage.kt

```diff
@@ -0,0 +1,40 @@
+package com.zetapush.library
+
+import android.content.Context
+import android.preference.PreferenceManager
+import java.util.*
+
+class KeyValueCredentialsStorage(
+    private val context: Context,
+    private val keyLogin: String,
+    private val keyPassword: String
+) : StorageCredentialsInterface {
+
+    override fun saveCredentials(login: String?, password: String?) {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+        val editor = sharedPreferences.edit()
+        editor.putString(keyLogin, login)
+        editor.putString(keyPassword, password)
+        editor.apply()
+    }
+
+    override val credentials: Map<String, String>
+        get() {
+            val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+            val login = sharedPreferences.getString(keyLogin, null)
+            val password = sharedPreferences.getString(keyPassword, null)
+
+            val credentials = HashMap<String, String>()
+            login?.let { login ->
+                credentials["login"] = login
```

Comment:
Constant? And maybe you should return a Credentials Object?

Reply:
I prefer manipulate an object than a Hashmap

Reply:
but if you want more data in credential object, you can add a field, i.e: "data" with a Map type. :)